### PR TITLE
Use Upstream Attestation Library

### PR DESCRIPTION
**Pull request by josephsweeney**

### CODE REVIEW

**Review by woodruffw — APPROVED** [review]
Nice, LGTM. 

I think one thing we want to consider here is re-enabling support for verify bottles of different arches/tags -- but we can do that with a follow-up, to keep the diff simple.

File: cmd/verify.rb [hunk 1]
```diff
@@ -68,46 +67,8 @@ def run
...
-                        ohai "Verified #{bottle.name} with tag #{bottle_tag} using backfill signature."
-                      end
-                    else
-                      opoo "Unable to verify #{bottle.name} with tag #{bottle_tag}. " \
-                           "Backfilled signature dated after last backfill date."
-                    end
-                  end
-                end
-              end
+              attestation = Homebrew::Attestation.check_core_attestation bottle
```
Comment (woodruffw) [inline cmd/verify.rb]: Nitpick: maybe catch the `AttestationError` that happens here and convert it into something a little more human-friendly?

**Comment by josephsweeney** [comment]
> I think one thing we want to consider here is re-enabling support for verify bottles of different arches/tags -- but we can do that with a follow-up, to keep the diff simple.

Unless I am missing something this is still possible by just passing in the flags? 

`brew verify --os=sonoma --arch=intel xz` as an example.

File: cmd/verify.rb (hunk 1 above)
Comment (josephsweeney) [reply cmd/verify.rb]: Done, I think `odie` is the correct option here?

Comment (josephsweeney) [reply cmd/verify.rb]: Actually went back to `opoo` because we don't actually die if there was an error.

**Comment by woodruffw** [comment]
> Unless I am missing something this is still possible by just passing in the flags?
> 
> `brew verify --os=sonoma --arch=intel xz` as an example.

Whoops, I misread the diff. You're right!

File: cmd/verify.rb (hunk 1 above)
Comment (woodruffw) [reply cmd/verify.rb]: Hmm, I think we want to change that behavior and actually die if there's an error (that way people can script with this subcommand). But we can iterate on that.

**Pull request merged**